src/AEG/AEGUtils.ts

```diff
@@ -0,0 +1,364 @@
+import {Ellipse} from "./Ellipse";
+import {Point} from "./Point";
+import {Rectangle} from "./Rectangle";
+
+/**
+ * Method that checks whether one shape overlaps another
+ * @param newShape The new shape that might be overlapping the existing shape
+ * @param existingShape The existing shape
+ * @returns True, if the new shape overlaps the existing shape. Else, false
+ */
+export function shapesOverlaps(
+    newShape: Rectangle | Ellipse,
+    existingShape: Rectangle | Ellipse
+): boolean {
+    if (newShape instanceof Rectangle) {
+        if (existingShape instanceof Rectangle) {
+            //For rectangle-rectangle, check if their edges intersect
+            return edgesIntersect(newShape, existingShape);
+        } else {
+            //For ellipse-rectangle collision, check if points on the ellipse are
+            //within the rectangle
+            const points: Point[] = getEllipsePoints(existingShape as Ellipse, 64);
+            for (let i = 0; i < points.length; i++) {
+                if (pointInRect(newShape as Rectangle, points[i])) {
+                    return true;
+                }
+            }
+            return false;
+        }
+    } else {
+        if (existingShape instanceof Rectangle) {
+            //For ellipse-rectangle collision, check if points on the ellipse are
+            //within the rectangle
+            const points: Point[] = getEllipsePoints(newShape as Ellipse, 64);
+            for (let i = 0; i < points.length; i++) {
+                if (pointInRect(existingShape as Rectangle, points[i])) {
+                    return true;
+                }
+            }
+            return false;
```

Comment:
This is essentially the same code, it should be abstracted into the same or a helper function should be made.